doc/api/net.md

```diff
@@ -78,19 +137,29 @@ server.listen(() => {
 });
 ```
 
-Don't call `server.address()` until the `'listening'` event has been emitted.
-
 ### server.close([callback])
 <!-- YAML
 added: v0.1.90
 -->
 
-Stops the server from accepting new connections and keeps existing
-connections. This function is asynchronous, the server is finally
-closed when all connections are ended and the server emits a [`'close'`][] event.
-The optional `callback` will be called once the `'close'` event occurs. Unlike
-that event, it will be called with an Error as its only argument if the server
-was not open when it was closed.
+* `callback` {Function}
+
+The `server.close()` method stops the server from accepting new connections
+while maintaining existing connections that have not yet completed. Once the
+existing connections have closed, the server will emit the [`'close'`][]
+event and call the optional `callback` function (if given) with `null` passed
+as the only argument.
+
+If `server.close()` is called while the server *is not currently listening for
+new connections*, the `callback` is called with an `Error` passed as the only
+argument.
+
+```js
+server.close((err) => {
+  if (err) throw err;
```

Comment:
I think its very unusual to treat this as a unrecoverable error, perhaps

```
(err) => {
if (err)
  console.log('server wasn't open!')
else
  console.log('server is now closed')
}
```

Generally we wouldn't distinguish between the two, because you wanted the server close, and it is now closed (whether it was open before or not).